src/hooks/useExtraSafePaddingBottomStyle.ts

```diff
@@ -0,0 +1,34 @@
+import {useEffect, useMemo, useState} from 'react';
+import {Keyboard} from 'react-native';
+import useStyledSafeAreaInsets from './useStyledSafeAreaInsets';
+import useThemeStyles from './useThemeStyles';
+
+const useExtraSafePaddingBottomStyle = () => {
+    const styles = useThemeStyles();
+    const [willKeyboardShow, setWillKeyboardShow] = useState(false);
```

Comment:
- I don't think we should we the function you mentioned, because it is not used in our app, just in test file.
- We already have the way to get the keyboard state, see this [file](https://github.com/Expensify/App/blob/main/src/components/withKeyboardState.tsx). But it use ```keyboardDidShow``` and ```keyboardDidHide``` events to check if the keyboard is shown or not. It works well but there is an issue: when the keyboard is shown (user can see the keyboard), the bottom button is still applied the ```padding-bottom: 20px``` for a while. So I use ```keyboardWillShow``` and ```keyboardWillHide``` to fix that issue.